src/http-data-source.ts

```diff
@@ -399,63 +418,53 @@ export abstract class HTTPDataSource<TContext = any> extends DataSource {
   }
 
   private async request<TResult = unknown>(request: Request): Promise<Response<TResult>> {
-    if (Object.keys(request.query).length > 0) {
-      request.path = request.path + '?' + this.buildQueryString(request.query)
+    const options: Request = {
+      ...this.globalRequestOptions,
```

Comment:
Not sure if this is the intended behavior, but as was working before globalOptions override local request options not the other way around :S.